packages/asu-core/src/auth/services/cas.service.ts

```diff
@@ -0,0 +1,77 @@
+import {singleton} from "tsyringe";
+import {AxiosProvider} from "../../http/providers/axios.provider";
+import {Axios} from "axios";
+import {APIGatewayProxyEvent, APIGatewayProxyResult} from "aws-lambda";
+import {UnauthenticatedError} from "../../errors/unauthenticated.error";
+import {lambdaRedirect} from "../../aws/lambda/response-handler";
+import {lambdaHandleError} from "../../aws/lambda/error-handler";
+import {EnvironmentVariable} from "../../enums/environment-variable";
+
+const CAS_HOST = 'https://weblogin.asu.edu/cas';
+
+@singleton()
+export class CasService {
+  private httpClient: Axios;
+  private authUser: string;
+
+  constructor(private axiosProvider: AxiosProvider) {
+  }
+
+  private async getHttpClient(): Promise<Axios> {
+    if (!this.httpClient) {
+      this.httpClient = this.axiosProvider.resolve({
+        baseURL: CAS_HOST
+      });
+    }
+
+    return this.httpClient;
+  }
+
+  public async authenticate(event: APIGatewayProxyEvent, callback: ()=>Promise<APIGatewayProxyResult>): Promise<APIGatewayProxyResult> {
+    // Bypass auth on dev
+    if (process.env[EnvironmentVariable.Stage] === 'dev') {
+      if (event.queryStringParameters && event.queryStringParameters.skip_auth === 'y') {
+        this.authUser = 'skipped_auth';
```

Comment:
Thinking we need to take in an asurite id as well so that getCasUser doesn't throw.